api/api/utils/photon.py

```diff
@@ -20,13 +21,51 @@ class UpstreamThumbnailException(APIException):
     default_code = "upstream_photon_failure"
 
 
+ALLOWED_TYPES = ["gif", "jpg", "jpeg", "png", "webp"]
+
 HEADERS = {
     "User-Agent": settings.OUTBOUND_USER_AGENT_TEMPLATE.format(
         purpose="ThumbnailGeneration"
     )
 }
 
 
+def _get_file_extension_from_url(image_url: str) -> str:
+    """Return the image extension if present in the URL."""
+    parsed = urlparse(image_url)
+    _, ext = splitext(parsed.path)
+    return ext[1:]  # remove the leading dot
+
+
+def check_image_type(image_url: str, media_obj) -> None:
+    cache = django_redis.get_redis_connection("default")
+    key = f"media:{media_obj.identifier}:thumb_type"
+
+    ext = _get_file_extension_from_url(image_url)
+
+    if not ext:
+        # If the extension is not present in the URL, try to get it from the redis cache
+        ext = cache.get(key)
+
+    if not ext:
+        # If the extension is still not present, try getting it from the content type
+        try:
+            response = requests.head(image_url)
+            response.raise_for_status()
+        except Exception as exc:
+            sentry_sdk.capture_exception(exc)
+            raise UpstreamThumbnailException(
+                "Failed to render thumbnail due to inability to check media "
+                f"type. {exc}"
+            )
+        else:
+            ext = response.headers["Content-Type"].split("/")[1]
```

Comment:
Would there ever be a case where `Content-Type` would not be in the `headers` of the response? Would we want to add some resiliency around that case?